uxarray/grid/neighbors.py

```diff
@@ -33,86 +38,151 @@ class KDTree:
 
     def __init__(self,
                  grid,
-                 tree_type: Optional[str] = "nodes",
-                 distance_metric="minkowski"):
+                 coordinates: Optional[str] = "nodes",
+                 coordinate_system="cartesian",
+                 distance_metric="minkowski",
+                 reconstruct=False):
 
         # Set up references
         self._source_grid = grid
-        self._tree_type = tree_type
+        self._coordinates = coordinates
+        self.coordinate_system = coordinate_system
         self.distance_metric = distance_metric
+        self.reconstruct = reconstruct
 
         self._tree_from_nodes = None
         self._tree_from_face_centers = None
+        self._tree_from_edge_centers = None
 
-        # Build the tree based on face centers or nodes
-        if tree_type == "nodes":
+        # Build the tree based on nodes, face centers, or edge centers
+        if coordinates == "nodes":
             self._tree_from_nodes = self._build_from_nodes()
             self._n_elements = self._source_grid.n_node
-        elif tree_type == "face centers":
+        elif coordinates == "face centers":
             self._tree_from_face_centers = self._build_from_face_centers()
             self._n_elements = self._source_grid.n_face
+        elif coordinates == "edge centers":
+            self._tree_from_edge_centers = self._build_from_edge_centers()
+            self._n_elements = self._source_grid.n_edge
         else:
             raise ValueError
 
     def _build_from_nodes(self):
         """Internal``sklearn.neighbors.KDTree`` constructed from corner
         nodes."""
-        if self._tree_from_nodes is None:
-            cart_coords = np.stack((self._source_grid.node_x.values,
-                                    self._source_grid.node_y.values,
-                                    self._source_grid.node_z.values),
-                                   axis=-1)
-            self._tree_from_nodes = SKKDTree(cart_coords,
+        if self._tree_from_nodes is None or self.reconstruct:
+
+            # Sets which values to use for the tree based on the coordinate_system
+            if self.coordinate_system == "cartesian":
+                coords = np.stack((self._source_grid.node_x.values,
+                                   self._source_grid.node_y.values,
+                                   self._source_grid.node_z.values),
+                                  axis=-1)
+
+            elif self.coordinate_system == "spherical":
+                coords = np.vstack(
+                    (deg2rad(self._source_grid.node_lat.values),
+                     deg2rad(self._source_grid.node_lon.values))).T
+
+            else:
+                raise TypeError
```

Comment:
Can we populate the various TypeErrors and ValueErrors with appropriate messages?